langserver/symbol.go

```diff
@@ -68,14 +72,50 @@ func queryJoin(s, e string) string {
 	return strings.TrimSpace(s + " " + e)
 }
 
+func adjustCase(q string) (qu Query, fields []string) {
+	FieldName := "CaseSensitive"
+	FieldNamePrefix := fmt.Sprintf("%s:", FieldName)
+
+	for _, field := range strings.Fields(q) {
+		// Check if the field is a filter like `CaseSensitive:true`.
+		if !strings.HasPrefix(field, FieldNamePrefix) {
+			continue
+		}
+
+		fieldVal := strings.TrimPrefix(field, FieldNamePrefix)
+		if caseSensitive, err := strconv.ParseBool(fieldVal); err != nil {
+			log.Printf("ParseQuery:CaseSensitive - parse error: %v", err)
+			break
+		} else {
+			qu.CaseSensitive = caseSensitive
+			break
+		}
+	}
+
+	// adjust the case of the fields or leave as-is
+	for _, field := range strings.Fields(q) {
+		if strings.HasPrefix(field, FieldNamePrefix) {
+			continue
+		}
+
+		if qu.CaseSensitive {
+			fields = append(fields, field)
+		} else {
+			fields = append(fields, strings.ToLower(field))
+		}
+	}
+
+	return qu, fields
+}
+
 // ParseQuery parses a user's raw query string and returns a
 // structured representation of the query.
 func ParseQuery(q string) (qu Query) {
-	// All queries are case insensitive.
-	q = strings.ToLower(q)
+	// All queries are case insensitive, unless otherwise
+	qu, fields := adjustCase(q)
```

Comment:
yeh, ok that's cleaner. thanks.
so effectively, `qu = strings.toLower(qu.String())`?